### PR TITLE
New updates for API workflow End to End 

### PR DESCRIPTION
**Pull request by eujinlok**
Hi @john-hawkins and @jessezwu  

Please accept my pull request for this new branch. This is the same python workflow API from 'retrain_python', but updated with recent updates from DataRobot. The workflow has been tested on a real usecase. I've documented the workflow and steps to reproduce in the Readme file in the new API_Workflow_EndtoEnd folder

Thanks
EJ

### CODE REVIEW

File: API_Workflow_EndtoEnd/.ipynb_checkpoints/1. Model Training-checkpoint.ipynb [hunk 1]
```diff
@@ -0,0 +1,128 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 1. Model Training \n",
+    "Using the API to train a model on DataRobot, and serealise the model  \n",
```
Comment (jessezwu) [inline API_Workflow_EndtoEnd/.ipynb_checkpoints/1. Model Training-checkpoint.ipynb]: typo :)

File: API_Workflow_EndtoEnd/4.Model_Monitoring_Management.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import sys
+import requests
+import json 
+
+with open('API_Workflow_EndtoEnd/Deployment_Creds.json') as f:
+    param = json.load(f)
+
+DATAROBOT_KEY = param['DATAROBOT_KEY']
+API_KEY = param['API_KEY']
+USERNAME = 'eu.jin.lok@datarobot.com'
```
Comment (jessezwu) [inline API_Workflow_EndtoEnd/4.Model_Monitoring_Management.py]: Probably remove this

**Review by john-hawkins — CHANGES_REQUESTED** [review]
Hey Eu Jin

Please note that this is a public repository.
You have your API Key and DataRobot Key in this commit.
This would allow anyone else to use your DataRobot account.

I usually add a dummy file with 'XXXXXXXXXXXXXXXX'
So that users know what to do.

Also, I strongly recommend adding any custom functions in a separate python file.
Notebooks are great for data scientists, but annoying for people who need to manage
deployed code.

Cheers
John

File: API_Workflow_EndtoEnd/4.Model_Monitoring_Management.py (hunk 2 above)
Comment (john-hawkins) [reply API_Workflow_EndtoEnd/4.Model_Monitoring_Management.py]: Jesse means your email address

**Comment by john-hawkins** [comment]
Eu Jin,

Note that you will need to remove these commits from your GitHub history.

Otherwise, someone nefarious could check the commit history and find your credentials.

Cheers
John

**Comment by eujinlok** [comment]
Thanks @john-hawkins  

My bad, I forgot to update the gitignore file to not commit the credentials. I've gone back and erased historical commits now. Let me know if you still see any trails of my creds anywhere 

Just the last one, what custom function are you referring to?

**Review by john-hawkins — APPROVED** [review]
Thanks Eu Jin,

Looks good, I don't see credentials in there anymore :-)

I will show you an example of what I meant by that other comment.

**Pull request merged**